plugins/bazaar/src/components/EntityBazaarInfoCard/EntityBazaarInfoCard.tsx

```diff
@@ -0,0 +1,345 @@
+    const response = await bazaarApi.getMembers(entity);
+    const dbMembers = response.data.map((obj: any) => {
+      const member: Member = {
+        userId: obj.user_id,
+        entityRef: obj.entity_ref,
+        joinDate: obj.join_date,
+      };
+
+      return member;
+    });
+
+    dbMembers.sort(sortMembers);
+
+    setMembers(dbMembers);
+    setIsMember(
+      dbMembers
+        .map((member: Member) => member.userId)
+        .indexOf(identity.getUserId()) >= 0,
+    );
+  };
+
+  const getMetadata = async () => {
+    const response = await bazaarApi.getMetadata(entity);
+
+    if (response.status !== 404) {
+      setIsBazaar(true);
+      const data = await response.json().then((resp: any) => resp.data);
+
+      setBazaarProject({
+        entityRef: data[0].entityRef,
+        name: data[0].name,
+        community: data[0].community,
+        announcement: data[0].announcement,
+        status: data[0].status,
+        updatedAt: data[0].updatedAt,
+      });
+    }
+  };
+
+  const { loading } = useAsync(async () => {
+    await getInitMemberStatus();
+    await getMetadata();
+  });
+
+  const onOpen = (event: React.SyntheticEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+    setPopoverOpen(true);
+  };
+
+  const closeEdit = () => {
+    setOpen(false);
+  };
+
+  const closeDelete = () => {
+    setOpenDelete(false);
+  };
+
+  const popoverCloseHandler = () => {
+    setPopoverOpen(false);
+  };
+
+  const handleMembersClick = async () => {
+    setIsMember(!isMember);
+
+    const newMember: Member = {
+      userId: identity.getUserId(),
+      entityRef: stringifyEntityRef(entity),
+      joinDate: new Date().toISOString(),
+    };
+
+    if (!isMember) {
+      setMembers((prevMembers: Member[]) => {
+        const newMembers: Member[] = [...prevMembers, newMember];
+        newMembers.sort(sortMembers);
+        return newMembers;
+      });
+      await bazaarApi.addMember(entity);
+    } else {
+      setMembers(
+        members.filter(
+          (member: Member) => member.userId !== identity.getUserId(),
+        ),
+      );
+      await bazaarApi.deleteMember(entity);
+    }
+  };
+
+  const links: IconLinkVerticalProps[] = [
+    {
+      label: 'Community',
+      icon: <ChatIcon />,
+      href: bazaarProject.community,
+      disabled: bazaarProject.community === '',
+    },
+    {
+      label: isMember ? 'Leave' : 'Join',
+      icon: isMember ? <ExitToAppIcon /> : <PersonAddIcon />,
+      href: '',
+      onClick: async () => {
+        handleMembersClick();
+      },
+    },
+  ];
+
+  if (loading) {
+    return <Progress />;
+  } else if (!isBazaar) {
+    return (
+      <Card>
+        <CardHeader title="Bazaar" style={{ paddingBottom: '1rem' }} />
+        <Divider />
+        <CardContent>
+          <Typography variant="body1">
+            This project is not in the Bazaar. Go to the{' '}
+            <Link className={classes.link} to="/bazaar">
```

Comment:
ideally this should have used the route ref that the plugin exposes, if the user mounted the plugin on a different route than specifically `/bazaar`